noir-projects/noir-contracts/contracts/claim_contract/src/main.nr

```diff
@@ -0,0 +1,39 @@
+contract Claim {
+    mod interfaces;
+
+    use dep::aztec::{
+        protocol_types::address::AztecAddress,
+        state_vars::PrivateImmutable,
+    };
+    use dep::field_note::field_note::FieldNote;
+    use dep::value_note::value_note::ValueNote;
+    use interfaces::Token;
+    use dep::aztec::history::note_inclusion::prove_note_inclusion;
+
+    struct Storage {
+      campaign_contract: PrivateImmutable<FieldNote>,
+      reward_token: PrivateImmutable<FieldNote>,
```

Comment:
I suspect this might be because you're keeping these notes private, and sharing the contents out-of-band instead of using a public immutable value (e.g. StableState) with broadcast. If that's the case, then I'd include a comment explaning that and to add the note to PXE.

I don't see however why this information can't be made public, which would make discovery etc much simpler.

Reply:
Well something like stable state would mean that you don't need to add any notes etc., so that definitely seems simpler. And it doesn't sound like making this information public would reveal much (anything), even in a real use case.

Reply:
Will address and use PublicImmutable.